src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs

```diff
@@ -16,127 +16,144 @@
-        }
+    private static (SourceText text, TextSpan semicolonSpan) DedentNamespace(
+        ParsedDocument document, string indentation, SyntaxAnnotation annotation, CancellationToken cancellationToken)
+    {
+        var syntaxTree = document.SyntaxTree;
+        var text = document.Text;
+        var root = document.Root;
 
-        private static (SourceText text, TextSpan semicolonSpan) DedentNamespace(
-            ParsedDocument document, string indentation, SyntaxAnnotation annotation, CancellationToken cancellationToken)
-        {
-            var syntaxTree = document.SyntaxTree;
-            var text = document.Text;
-            var root = document.Root;
+        var fileScopedNamespace = (FileScopedNamespaceDeclarationSyntax)root.GetAnnotatedNodes(annotation).Single();
+        var semicolonLine = text.Lines.GetLineFromPosition(fileScopedNamespace.SemicolonToken.SpanStart).LineNumber;
 
-            var fileScopedNamespace = (FileScopedNamespaceDeclarationSyntax)root.GetAnnotatedNodes(annotation).Single();
-            var semicolonLine = text.Lines.GetLineFromPosition(fileScopedNamespace.SemicolonToken.SpanStart).LineNumber;
+        // Cache what we compute so we don't have to recompute it for every line of a raw string literal.
+        (SyntaxToken stringLiteral, int closeTerminatorIndentationLength) lastRawStringLiteralData = default;
 
-            using var _ = ArrayBuilder<TextChange>.GetInstance(out var changes);
-            for (var line = semicolonLine + 1; line < text.Lines.Count; line++)
-                changes.AddIfNotNull(TryDedentLine(syntaxTree, text, indentation, text.Lines[line], cancellationToken));
+        using var _ = ArrayBuilder<TextChange>.GetInstance(out var changes);
+        for (var line = semicolonLine + 1; line < text.Lines.Count; line++)
+            changes.AddIfNotNull(TryDedentLine(text.Lines[line]));
 
-            var dedentedText = text.WithChanges(changes);
-            return (dedentedText, fileScopedNamespace.SemicolonToken.Span);
-        }
+        var dedentedText = text.WithChanges(changes);
+        return (dedentedText, fileScopedNamespace.SemicolonToken.Span);
 
-        private static TextChange? TryDedentLine(
-            SyntaxTree tree, SourceText text, string indentation, TextLine textLine, CancellationToken cancellationToken)
+        TextChange? TryDedentLine(TextLine textLine)
         {
             // if this line is inside a string-literal or interpolated-text-content, then we definitely do not want to
             // touch what is inside there.  Note: this will not apply to raw-string literals, which can potentially be
             // dedented safely depending on the position of their close terminator.
-            if (tree.IsEntirelyWithinStringLiteral(textLine.Span.Start, cancellationToken))
-                return null;
+            if (syntaxTree.IsEntirelyWithinStringLiteral(textLine.Span.Start, out var stringLiteral, cancellationToken))
```

Comment:
hmm, the issue was mentioned in the comment but not actually handled

Reply:
not sure what you mean :)  the handling of raw-string-literals is what the new code is below :)

Reply:
> Note: this will not apply to raw-string literals, which can potentially be dedented safely depending on the position of their close terminator.

I was referring to this comment above. I guess this was written before raw string literals became available?